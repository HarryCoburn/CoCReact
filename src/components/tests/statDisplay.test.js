import React from "react";
import renderer from "react-test-renderer";
import StatDisplay from "../statDisplay";

const dummy = {
  ids: ["strength", "toughness"],
  stats: {
    byID: {
      strength: {
        name: "Strength",
        value: 50,
        max: 100
      },
      toughness: {
        name: "Toughness",
        value: 75,
        max: 100
      }
    }
  }
};

describe("StatDisplay Component", () => {
  it("renders correctly", () => {
    const component = renderer.create(
      <StatDisplay ids={dummy.ids} stats={dummy.stats} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
