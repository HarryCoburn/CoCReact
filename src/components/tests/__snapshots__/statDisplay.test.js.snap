// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatDisplay Component renders correctly 1`] = `
Array [
  <div
    className="statFrame"
  >
    <div
      className="statGradient"
    >
      <p
        className="statLabel"
      >
        Strength
        : 
        50
      </p>
    </div>
  </div>,
  <div
    className="statFrame"
  >
    <div
      className="statGradient"
    >
      <p
        className="statLabel"
      >
        Toughness
        : 
        75
      </p>
    </div>
  </div>,
]
`;
